sdk/eventhub/Azure.Messaging.EventHubs/src/TrackOneClient/EventDataSender.cs

```diff
@@ -26,9 +26,12 @@ protected EventDataSender(EventHubClient eventHubClient, string partitionId)
         public async Task SendAsync(IEnumerable<EventData> eventDatas, string partitionKey)
         {
             int count = ValidateEvents(eventDatas);
+            var diagnosticPartitionKey = String.IsNullOrEmpty(partitionKey) ?
```

Comment:
Not sure that you want to do this;  the names are way too similar, but they represent very different things and aren't interchangeable.

Id == The unique identifier of a partition.
Key ==A user-provided value to be hashed and then mapped to a partition.

It's valid (and expected) for there to be no partition key in many cases.

Reply:
I agree with you, and I was pretty confused as well at first, but digging through Track One code I was able to understand what was going on.

The `PartitionKey` tag and payload string (in Diagnostics events) are used in two situations in Track One:
- It holds the Partition Key value when we send to no specific partition (using `Client.SendAsync`). It can be null if no Partition Key is specified. That's nice and it makes sense.
- It holds the Partition Id value when we send to send to a specific partition (using `PartitionSender.SendAsync`). This is very misleading. [You can check the Track One code here](https://github.com/Azure/azure-sdk-for-net/blob/9904eab625e9f73827ae2644b78742d9dc00b53e/sdk/eventhub/Microsoft.Azure.EventHubs/src/PartitionSender.cs#L135).

I simply tried to mimic this behavior since we are not using `PartitionSender` internally.

Reply:
Is this what track one is doing?   Using key when it's not a partition sender and id when it is?  If so, then I'm good with the approach.... but we should rename the variable to something like `activePartitionRouting` or something that doesn't seem like we're conflating the two.

Reply:
Yes. I'll update the variable. Do you think we should update the "PartitionKey" tag and payload identifiers as well or leave them for the actual Track Two implementation?

Reply:
I'll leave that to your discretion.  There's no real reason not to update them unless it's risky or time consuming.   But, update or leave a `//TODO:` is good by me.

Reply:
Just updated everything. I'll push it with other changes in a few moments.